### PR TITLE
[ASCII-2584] Migrating CoreAgent to use IPC cert

### PR DESCRIPTION
**Pull request by misteriaud**
### What does this PR do?
This PR replaces the self-signed certificate used in the Core Agent with the new Agent-wide IPC certificate. This change will allow Agents to identify this API as legitimate when they act as a client.

The certificate generation/retrieval implementation have been made in the following PR:
- #31838

### Motivation
This PR takes place in a plan of Agent security IPC improvement. 

### Describe how you validated your changes

Core Agent CMD_API is a core block of the core Agent and almost all Agent e2e tests rely on it. They should fail if the current PR introduce bugs.

### CODE REVIEW

File: comp/api/api/apiimpl/server.go [hunk 1]
```diff
@@ -47,43 +48,23 @@ func (server *apiServer) startServers() error {
...
 		tmf,
 		server.cfg,
 	); err != nil {
 		return fmt.Errorf("unable to start CMD API server: %v", err)
 	}
 
 	// start the IPC server
-	if ipcServerEnabled {
-		if err := server.startIPCServer(ipcServerHostPort, tlsConfig(), tmf); err != nil {
+	if ipcServerPort := server.cfg.GetInt("agent_ipc.port"); ipcServerPort != 0 {
```
Comment (louis-cqrl) [inline comp/api/api/apiimpl/server.go:63]: What happen if `ipcServerPort == 0` ?

Comment (misteriaud) [reply comp/api/api/apiimpl/server.go:63]: if `ipcServerPort` is set to 0 we consider that it should not be started.

**Review by louis-cqrl — APPROVED** [review]
LGTM

**Comment by misteriaud** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Pull request merged**